.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,22 +1,40 @@
-# PR Type
+# 🔘 PR Type
 What kind of change does this PR introduce?
 
 <!-- Please check the one that applies to this PR using "x". -->
 
 - [ ] Bugfix
 - [ ] Enhancement
 - [ ] Code style update (formatting, local variables)
-- [ ] Refactoring (no functional changes, no api changes)
+- [ ] Refactoring (no functional changes, no API changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Documentation content changes
 - [ ] Other... Please describe:
 
+# 📜 Background
+Provide a brief explanation of why this pull request is needed. Include the problem you are solving or the functionality you are adding. Reference any related issues.
 
-# What is the current behavior?
-(Optional) Resolved: #`Insert issue number`
+Issue Link: #[issue_number]
+Context Link (if applicable):
 
+# 🔄 Changes
+Enumerate the changes made in this pull request, detailing what has been modified, added, or removed. Include technical details and implications if necessary.
```

Comment:
Would be good if we could also add what effect this could have within the service so that we think about and document what parts need to be QA'd in case of side effects.